source/core/Ky.ts

```diff
@@ -105,43 +105,54 @@ export class Ky {
 				throw new RangeError(`The \`timeout\` option cannot be greater than ${maxSafeTimeout}`);
 			}
 
-			await delay(1);
-			let response = await ky._fetch();
-
-			for (const hook of ky._options.hooks.afterResponse) {
-				// eslint-disable-next-line no-await-in-loop
-				const modifiedResponse = await hook(
-					ky.request,
-					ky._options as NormalizedOptions,
-					ky._decorateResponse(response.clone())
-				);
-
-				if (modifiedResponse instanceof globalThis.Response) {
-					response = modifiedResponse;
-				}
-			}
+			return new Promise<Response>((resolve, reject) => {
+				// Fix #345: We must keep setTimeout() here instead of using delay(), because new relic cannot handle parallel promise correctly.
+				setTimeout(async () => {
```

Comment:
The major change is just remove the `delay(1)` and perform the delay by a setTimeout.